app/forms/state_file/esign_declaration_form.rb

```diff
@@ -0,0 +1,26 @@
+module StateFile
+  class EsignDeclarationForm < QuestionsForm
+    set_attributes_for :intake,
+                       :primary_esigned,
+                       :spouse_esigned
+
+    validates :primary_esigned, acceptance: { accept: 'yes', message: I18n.t("views.ctc.questions.confirm_legal.error") }
+    validates :spouse_esigned, acceptance: { accept: 'yes', message: I18n.t("views.ctc.questions.confirm_legal.error") }, if: -> { @intake.filing_status_mfj? }
+
+    def save
+      attrs = @intake.filing_status_mfj? ? attributes_for(:intake) : attributes_for(:intake).except(:spouse_esigned)
+      @intake.update!(attrs)
+      @intake.touch(:primary_esigned_at) if @intake.primary_esigned_yes?
+      @intake.touch(:spouse_esigned_at) if @intake.spouse_esigned_yes? && @intake.filing_status_mfj?
+
+      # Submits return
+      efile_submission = EfileSubmission.create!(
+        data_source: @intake,
+      )
+      if Rails.env.development? || Rails.env.test?
+        efile_submission.transition_to(:preparing)
+      end
```

Comment:
Why is this scoped to the test & dev environments? Is that to fake the behavior of getting a response back from the IRS?

Reply:
I just moved this from the previous controller. It just looks like new code because the controller name has changed but it has already been code reviewed and I don't have context on it @bengolder